source/obs/obs-tools.hpp

```diff
@@ -23,7 +23,7 @@
 
 namespace streamfx::obs {
 	namespace tools {
-		bool source_find_source(::streamfx::obs::source haystack, ::streamfx::obs::source needle);
+		bool source_find_source(const ::streamfx::obs::source& haystack, const ::streamfx::obs::source& needle);
```

Comment:
This change may have unintended side effects that will not be immediately obvious. The choice of passing objects as copies/moves instead of by-reference was intentional to ensure that the OBS internal reference counting still works as intended.